Storage/Rust/RustPlaces.swift

```diff
@@ -408,29 +408,48 @@ public class RustPlaces: BookmarksHandler, HistoryMetadataObserver {
 
 // MARK: History APIs
 
-extension VisitTransition {
-    public static func fromVisitType(visitType: VisitType) -> Self {
-        switch visitType {
-        case .unknown:
-            return VisitTransition.link
-        case .link:
-            return VisitTransition.link
-        case .typed:
-            return VisitTransition.typed
-        case .bookmark:
-            return VisitTransition.bookmark
-        case .embed:
-            return VisitTransition.embed
-        case .permanentRedirect:
-            return VisitTransition.redirectPermanent
-        case .temporaryRedirect:
-            return VisitTransition.redirectTemporary
-        case .download:
-            return VisitTransition.download
-        case .framedLink:
-            return VisitTransition.framedLink
-        case .recentlyClosed:
-            return VisitTransition.link
+// WKWebView has these:
+/*
+WKNavigationTypeLinkActivated,
+WKNavigationTypeFormSubmitted,
+WKNavigationTypeBackForward,
+WKNavigationTypeReload,
+WKNavigationTypeFormResubmitted,
+WKNavigationTypeOther = -1,
+*/
+
+// The previous VisitType started from .unknown at 1 which means everything is off
+// from desktop and rust by 1. 
+extension VisitType {
+    public static func fromRawValue(rawValue: Int?) -> Self {
```

Comment:
This whole extension and `VisitType` is weird because of Uniffi. I synced with Sammy and he explained on why we have to write this method.

@skhamis lets add a comment and come back to it at later time, r+ for now.